### PR TITLE
Make PGI CI conforming

### PR DESCRIPTION
**Pull request by masterleinad**

### CODE REVIEW

File: .jenkins [hunk 1]
```diff
@@ -193,6 +193,7 @@ pipeline {
                                   -D CMAKE_BUILD_TYPE=Debug \
                                   -D CMAKE_CXX_COMPILER_LAUNCHER=ccache \
                                   -D CMAKE_CXX_COMPILER=pgc++ \
+                                  -D CMAKE_CXX_FLAGS="-Wall" \
```
Comment (Rombur) [inline .jenkins]: Where did you find that flag? It's not in the [documentation](https://www.pgroup.com/resources/docs/19.4/x86/pgi-user-guide/index.htm)

Comment (masterleinad) [reply .jenkins]: I was removing everything that didn't work. Not sure if it has any effect. At least the compiler output didn't change. Replaced now with `-Minform=warn`. I don't see anything either, but at least it is documented (https://www.pgroup.com/resources/docs/19.4/x86/pgi-ref-guide/index.htm#cmdln-m-opts-misc__Minform).

File: .jenkins [hunk 2]
```diff
@@ -186,13 +186,32 @@ pipeline {
...
                         sh 'ccache --zero-stats'
-                        sh 'echo $PATH && rm -rf build && mkdir -p build && cd build && cmake -D CMAKE_BUILD_TYPE=Debug -D CMAKE_CXX_COMPILER_LAUNCHER=ccache -D CMAKE_CXX_COMPILER=pgc++ -D CMAKE_PREFIX_PATH="$KOKKOS_DIR;$BOOST_DIR;$BENCHMARK_DIR" -D ARBORX_ENABLE_MPI=ON -D MPIEXEC_PREFLAGS="--allow-run-as-root" .. && make -j8 VERBOSE=1 && ctest --no-compress-output -T Test'
+                        sh 'rm -rf build && mkdir -p build'
+                        dir('build') {
+                            sh '''
+                              cmake \
+                                  -D CMAKE_BUILD_TYPE=Debug \
+                                  -D CMAKE_CXX_COMPILER_LAUNCHER=ccache \
+                                  -D CMAKE_CXX_COMPILER=pgc++ \
+                                  -D CMAKE_CXX_FLAGS="-Minform=warn" \
```
Comment (Rombur) [inline .jenkins]: I am not sure what this flags does. The documentation is very confusing. Should we use `warn`:
```
warn
    instructs the compiler to display warning, severe and fatal error messages.
```
or `inform`:
```
inform
    instructs the compiler to display all error messages (inform, warn, severe and fatal).
```

**Comment by Rombur** [comment]
retest this please

**Comment by masterleinad** [comment]
Why did you close?

**Comment by dalg24** [comment]
Re-open if you like.

**Comment by masterleinad** [comment]
So you are saying we should just forget about the warning flag?

**Comment by dalg24** [comment]
GIT accident

**Comment by dalg24** [comment]
Dammit can't reopen

**Pull request closed without merging**